### PR TITLE
Hero pattern

### LINKED ISSUE #107: Block Patterns - Hero
**Description**

🛠️ The goal of this issue is to track the implementation of the hero pattern.
🎨 [Figma](https://www.figma.com/file/AlYr03vh4dVimwYwQkTdf6/Twenty-Twenty-Four?type=design&node-id=536%3A1918&mode=design&t=LHSPHKUrPq556l59-1)

Hero

### PR DESCRIPTION
**Pull request by maneshtimilsina**

### CODE REVIEW

**Comment by kafleg** [comment]
Looks good to me. But, I see this pattern is already in the home.php pattern. But I think creating a separate pattern for each section is good.

**Review by kafleg — APPROVED** [review]

**Review by MaggieCabrera — CHANGES_REQUESTED** [review]
Same comment as in your previous PR, let's move the markup from the original template without modifying any of the design elements (just adding translation and image urls) and replace with the pattern block in the original template

**Comment by maneshtimilsina** [comment]
@MaggieCabrera changes are done and updated as per your instructions. Please verify it and let me know if there is anything left to do.

**Comment by maneshtimilsina** [comment]
@MaggieCabrera I haven't replaced the content of `home.php` to avoid merge conflict. I think it will be better to replace the codes of `home.php` with patterns once all patterns are approved and merged to the trunk.

**Comment by richtabor** [comment]
>  I think it will be better to replace the codes of home.php with patterns once all patterns are approved and merged to the trunk.

Agreed. We'll probably consolidate once the individual patterns are cleaned up.

File: patterns/hero.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Title: Hero
+ * Slug: twentytwentyfour/hero
+ * Categories: banner
+ */
+?>
+
+<!-- wp:group {"align":"wide","style":{"spacing":{"padding":{"top":"6vh","bottom":"6vh"}}},"layout":{"type":"constrained"}} -->
```
Comment (richtabor) [inline patterns/hero.php]: Instead of arbitrary values, let's use the spacing presets for paddings throughout.

File: patterns/hero.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<div class="wp-block-group" style="margin-top:0;margin-bottom:0;padding-top:var(--wp--preset--spacing--50);padding-bottom:var(--wp--preset--spacing--80)"><!-- wp:heading {"textAlign":"center"} -->
+<h2 class="wp-block-heading has-text-align-center"><?php echo esc_html_x( 'A commitment to innovation and sustainability', 'Heading of the hero section', 'twentytwentyfour' ); ?></h2>
+<!-- /wp:heading -->
+
+<!-- wp:paragraph {"align":"center","style":{"layout":{"selfStretch":"fit","flexSize":null},"spacing":{"padding":{"right":"0","left":"0"}}},"fontSize":"small"} -->
+<p class="has-text-align-center has-small-font-size" style="padding-right:0;padding-left:0"><?php echo esc_html_x( 'Études is a pioneering firm that seamlessly merges creativity and functionality to redefine architectural excellence.', 'Content of the hero section', 'twentytwentyfour' ); ?></p>
```
Comment (richtabor) [inline patterns/hero.php]: Not sure if removing the padding is necessary here.

File: patterns/hero.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+<!-- /wp:paragraph -->
+
+<!-- wp:buttons {"layout":{"type":"flex","justifyContent":"center"}} -->
+<div class="wp-block-buttons"><!-- wp:button -->
+<div class="wp-block-button"><a class="wp-block-button__link wp-element-button"><?php echo esc_html_x( 'Learn More', 'Button of the hero section', 'twentytwentyfour' ); ?></a></div>
+<!-- /wp:button --></div>
+<!-- /wp:buttons --></div>
+<!-- /wp:group -->
+
+<!-- wp:image {"align":"wide","id":369,"sizeSlug":"full","linkDestination":"none"} -->
```
Comment (richtabor) [inline patterns/hero.php]: We should remove the ID attribute and class for images.

**Review by richtabor — DISMISSED** [review]

**Comment by richtabor** [comment]
This is what I'm seeing? 

CleanShot 2023-08-30 at 14 00 38

File: patterns/hero.php (hunk 1 above)
Comment (richtabor) [inline patterns/hero.php]: Let's also make it fullwidth.

**Comment by richtabor** [comment]
Let's try using spacer blocks (with preset, not pixels) instead of block gap for these here. Its easier for people to grab the space and adjust, rather than finding the block gap control: 

CleanShot 2023-08-30 at 14 01 25

**Comment by MaggieCabrera** [comment]
@maneshtimilsina are you working on the changes to this PR?

**Comment by maneshtimilsina** [comment]
@MaggieCabrera Yes, I will push my changes until Tuesday.

**Comment by maneshtimilsina** [comment]
@MaggieCabrera I have made the changes according to the suggestions of @richtabor and pushed the changes. @richtabor Can you please check it and let me know if you have any suggestions?

File: patterns/hero.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+<p class="has-text-align-center"><?php echo esc_html_x( 'Études is a pioneering firm that seamlessly merges creativity and functionality to redefine architectural excellence.', 'Content of the hero section', 'twentytwentyfour' ); ?></p>
+<!-- /wp:paragraph -->
+
+<!-- wp:spacer {"height":"0px","style":{"layout":{"flexSize":"1.25rem","selfStretch":"fixed"}}} -->
+<div style="height:0px" aria-hidden="true" class="wp-block-spacer"></div>
+<!-- /wp:spacer -->
+
+<!-- wp:buttons {"layout":{"type":"flex","justifyContent":"center"}} -->
+<div class="wp-block-buttons"><!-- wp:button -->
+<div class="wp-block-button"><a class="wp-block-button__link wp-element-button"><?php echo esc_html_x( 'Learn More', 'Button of the hero section', 'twentytwentyfour' ); ?></a></div>
```
Comment (MaggieCabrera) [inline patterns/hero.php]:
```suggestion
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button"><?php echo esc_html_x( 'Learn More', 'Button\'s label of the hero section', 'twentytwentyfour' ); ?></a></div>
```

**Review by MaggieCabrera — APPROVED** [review]
This is looking great, thanks!

**Review by MaggieCabrera — APPROVED** [review]

**Pull request merged**